sdk/generic-sdk-core/generic-core/src/main/java/com/generic/core/http/models/EventStreamListener.java

```diff
@@ -0,0 +1,47 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.generic.core.http.models;
+
+/**
+ * Interface to implement event stream listeners for handling {@link ServerSentEvent}.
+ */
+public interface EventStreamListener {
```

Comment:
Call this `ServerSentEventsListener`. `EventStreamListener` sounds like a more general-purpose type that can handle any event stream.